### ISSUE TITLE
Ineffective asserts in test_sockets_manager.py

### ISSUE THREAD

**Issue by matrss** [Open-MSS/MSS#2120]
In `test_join_collaborator_to_operation` on line 127 a Permission object is created and then asserts are used to check that its attributes match what was used to create it. This is essentially a no-op. See: https://github.com/Open-MSS/MSS/blob/788d10730adc45ff791a90b042347a800912bc42/tests/_test_mscolab/test_sockets_manager.py#L122-L130 

Instead, I think `perms` should be fetched from the database by matching against u_id and op_id, e.g. something like:
```python
Permission.query.filter_by(u_id=self.anotheruser.id, op_id=operation.id).all()
```

This applies to most or all uses of `Permission` in that file.

I found this when I tried to get `test_remove_collaborator_from_operation` to work. I don't see how these database entries would actually be generated though.

**Comment by ReimarBauer** [comment]
The test should show an other user than the creator joined as collaborator by a socket event operation 
The database entries are created by the join_collaborator_to_operation  event.


For me the test output looks like this.
```
======================== 1 passed, 7 warnings in 2.11s ========================
PASSED [100%]Database has been reset successfully!
Server initialized for eventlet.
Server initialized for eventlet.
ytkoWH4Mi-6j8ngbAAAA: Sending packet OPEN data {'sid': 'ytkoWH4Mi-6j8ngbAAAA', 'upgrades': ['websocket'], 'pingTimeout': 20000, 'pingInterval': 25000}
ytkoWH4Mi-6j8ngbAAAA: Received packet MESSAGE data 0{}
ytkoWH4Mi-6j8ngbAAAA: Sending packet MESSAGE data 0{"sid":"B7bFuYXW9EW_X98zAAAB"}
ytkoWH4Mi-6j8ngbAAAA: Received packet MESSAGE data 2["connect"]
received event "connect" from B7bFuYXW9EW_X98zAAAB [/]
```

a breakpoint on the asserts shows also in debugger the perms object

`<Permission u_id: 2, op_id:2, access_level: collaborator>`

The test in the old setup succeeds ;)

The query can become improved ;)

**Comment by matrss** [comment]
> The database entries are created by the join_collaborator_to_operation event.

That does not seem to happen. I manually checked the database at the end of the test and it only contains this:
```
sqlite> select * from permissions;
1|1|1|admin
```

> The test in the old setup succeeds ;)

It succeeds because
```
perms = Permission(self.anotheruser.id, operation.id, "collaborator") 
assert perms.op_id == operation.id 
assert perms.u_id == self.anotheruser.id 
assert perms.access_level == "collaborator"
```
is effectively an `assert True`, that is my point. It creates a Permission object and then immediately tests that its attributes are equal to those that were passed to its constructor. This does not touch the database at all.

> The query can become improved ;)

There is no query, that is the issue.

**Comment by ReimarBauer** [comment]
I was blind, yes you are correct.

**Comment by anj20** [comment]
I believe the same issue occurs in these two functions as well, where a new Permission object is created in memory, but the database is not queried to check for an existing permission.

https://github.com/Open-MSS/MSS/blob/788d10730adc45ff791a90b042347a800912bc42/tests/_test_mscolab/test_sockets_manager.py#L109-L120

https://github.com/Open-MSS/MSS/blob/788d10730adc45ff791a90b042347a800912bc42/tests/_test_mscolab/test_sockets_manager.py#L132-L143

**Comment by matrss** [comment]
Yes exactly, this is an issue in multiple places, at least in that file.
